File: android/src/main/java/com/lib/agnoreactnative/AgnoPlayBridgeModule.java
package com.lib.agnoreactnative;

import androidx.annotation.NonNull;
import android.view.View;
import com.egeniq.agno.agnoplayer.player.AgnoPlayer;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
public class AgnoPlayBridgeModule extends ReactContextBaseJavaModule {
    private final ReactApplicationContext reactContext;
    private int viewId = View.NO_ID;
    public AgnoPlayBridgeModule(ReactApplicationContext reactContext) {
        super(reactContext);
        this.reactContext = reactContext;
        AgnoPlayer.Companion.initialize(getReactApplicationContext());
    }

    @NonNull
    @Override
    public String getName() {
        return "AgnoPlay";
    }
}

File: android/src/main/java/com/lib/agnoreactnative/AgnoPlayBridgeModule.kt
package com.lib.agnoreactnative

import android.annotation.SuppressLint
import android.content.pm.ActivityInfo
import androidx.annotation.NonNull
import com.egeniq.agno.agnoplayer.player.AgnoPlayer
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod

class AgnoPlayBridgeModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

    init {
        AgnoPlayer.initialize(reactApplicationContext)
    }

    @SuppressLint("SourceLockedOrientationActivity")
    @ReactMethod
    fun lockToPortrait() {
        currentActivity?.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
    }

    @SuppressLint("SourceLockedOrientationActivity")
    @ReactMethod
    fun lockToLandscape() {
        currentActivity?.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
    }

    @NonNull
    override fun getName(): String {
        return "AgnoPlay"
    }
}
